dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java

```diff
@@ -66,7 +67,14 @@ public void setProxyFactory(ProxyFactory proxyFactory) {
         final String uri = serviceKey(invoker.getUrl());
         Exporter<T> exporter = (Exporter<T>) exporterMap.get(uri);
         if (exporter != null) {
-            return exporter;
+            // When modifying the configuration through override, you need to re-expose the newly modified service.
+            if ("hessian".equals(invoker.getUrl().getProtocol())) {

```

Comment:
We should not add special check for "hessian". It looks like we should compare URL for all proxy protocol when the cache is hit.